src/controllers/UserController.js

```diff
@@ -1,166 +1,440 @@
+        data: {},
+        message: "Something went wrong",
+      });
+    }
+
+    return res.status(200).json({
+      status: true,
+      data: products,
+      message: "Successful",
+    });
+    //send email to user
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+//update seller products
+exports.updateMyProducts = async (req, res) => {
+  try {
+    //store user details to database.
+    const { quantity, price, product_id, user_id } = req.body;
+
+    //store user details to database.
+    const product = await updateSellerProduct(
+      quantity,
+      price,
+      product_id,
+      user_id
+    );
+
+    if (!product) {
+      return res.status(400).json({
+        status: false,
+        data: {},
+        message: "Something went wrong",
+      });
+    }
+
+    return res.status(200).json({
+      status: true,
+      data: product,
+      message: "Successful",
+    });
+    //send email to user
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+// for buyers or even sellers to view all products available for sale
+exports.viewAllProducts = async (req, res) => {
+  try {
+    //store user details to database.
+    const products = await getAllProducts();
+    if (!products) {
+      return res.status(400).json({
+        status: false,
+        data: {},
+        message: "Something went wrong",
+      });
+    }
+
+    return res.status(200).json({
+      status: true,
+      data: products,
+      message: "Successful",
+    });
+    //send email to user
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+// for buyers or even sellers to view all products category available for sale
+exports.viewAllCategoryProducts = async (req, res) => {
+  try {
+    const { category_id } = req.params;
+    //store user details to database.
+    const products = await getCategoryProducts(category_id);
+    if (!products) {
+      return res.status(400).json({
+        status: false,
+        data: {},
+        message: "Something went wrong",
+      });
+    }
+
+    return res.status(200).json({
+      status: true,
+      data: products,
+      message: "Successful",
+    });
+    //send email to user
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+// for buyers or even sellers to view a single product available for sale
+exports.viewProduct = async (req, res) => {
+  try {
+    const { product_id } = req.params;
+    //store user details to database.
+    const product = await getProduct(product_id);
+    if (!product) {
+      return res.status(400).json({
+        status: false,
+        data: {},
+        message: "Something went wrong",
+      });
+    }
+
+    return res.status(200).json({
+      status: true,
+      data: product,
+      message: "Successful",
+    });
+    //send email to user
+  } catch (error) {
+    console.error(error);
+  }
+};
+exports.addToCart = async (req, res) => {
+  try {
+    const { user_id, productName, productId, quantity, price } = req.body;
```

Comment:
user_id is not really necessary when adding to cart. You can get it from req.user.id